custom_components/tahoma/binary_sensor.py

```diff
@@ -60,6 +60,12 @@ def device_class(self):
 
     def update(self):
         """Update the state."""
+        exec_queue = self.controller.get_current_executions()
```

Comment:
This code is the same in all implementations? Should we move it to the `tahoma_device` and call this function from here?

Reply:
It could take this form:
```
    def should_wait(self):
        """Wait for actions to finish (this would be in tahoma_device"""
        exec_queue = self.controller.get_current_executions()
        self._exec_queue = [e for e in self._exec_queue if e in exec_queue]
        return True if self._exec_queue else False

    def update(self):
        """Update the state."""
        if self.should_wait():
            self.schedule_update_ha_state(True)
            return
        # continue update
```